lib/ObjWriter/objwriter.cpp

```diff
@@ -420,6 +420,53 @@ extern "C" void EmitFrameInfo(ObjectWriter *OW, const char *FunctionName,
                 4);
 }
 
+extern "C" void EmitCFIStart(ObjectWriter *OW, int Offset) {
+  assert(OW && "ObjWriter is null");
+  assert(!OW->frameOpened && "frame should be closed before CFIStart");
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = *AsmPrinter->OutStreamer;
+
+  OST.EmitCFIStartProc(false);
+  OW->frameOpened = true;
+}
+
+extern "C" void EmitCFIEnd(ObjectWriter *OW, int Offset) {
+  assert(OW && "ObjWriter is null");
+  assert(OW->frameOpened && "frame should be opened before CFIEnd");
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = *AsmPrinter->OutStreamer;
+
+  OST.EmitCFIEndProc();
+  OW->frameOpened = false;
+}
+
+extern "C" void EmitCFIBlob(ObjectWriter *OW, int Offset, const char *Blob) {
```

Comment:
I might call this "EmitCFICode" instead of "EmitCFIBlob", then. IMO "EmitCFIBlob" implies that this function may process multiple CFI instructions.